cmd/kubeadm/app/util/staticpod/utils.go

```diff
@@ -140,3 +144,62 @@ func WriteStaticPodToDisk(componentName, manifestDir string, pod v1.Pod) error {
 
 	return nil
 }
+
+// GetProbeAddress returns an IP address or 127.0.0.1 to use for liveness probes
+// in static pod manifests. For etcd, GetProbeAddress will use the first
+// listen-client-url if multiple urls are provided. If the listen-client-url
+// contains a hostname instead of an IP address, GetProbeAddress will use the
+// local resolver for name resolution or 127.0.0.1 if the name can not be resolved.
+func GetProbeAddress(cfg *kubeadmapi.MasterConfiguration, component string) string {
+	switch {
+	case component == kubeadmconstants.KubeAPIServer:
+		if cfg.API.AdvertiseAddress != "" {
+			return cfg.API.AdvertiseAddress
+		}
+	case component == kubeadmconstants.KubeScheduler:
+		if cfg.SchedulerExtraArgs != nil {
+			addr, exists := cfg.SchedulerExtraArgs["address"]
+			if exists {
+				return addr
+			}
+		}
+	case component == kubeadmconstants.KubeControllerManager:
+		if cfg.ControllerManagerExtraArgs != nil {
+			addr, exists := cfg.ControllerManagerExtraArgs["address"]
+			if exists {
+				return addr
+			}
+		}
+	case component == kubeadmconstants.Etcd:
+		if cfg.Etcd.ExtraArgs != nil {
+			arg, exists := cfg.Etcd.ExtraArgs["listen-client-urls"]
+			if exists {
+				parsedURL, err := url.Parse(arg)
```

Comment:
what happens if I provide `http://10.0.0.1:2379,http://10.0.0.2:2379` for `listen-client-urls`? you're parsing the whole thing, right? we should probably split first

Reply:
since this doesn't look right, it may also indicate we're not testing this part correctly

Reply:
Yes, in your example `parsedURL := http://10.0.0.1:2379,http://10.0.0.2:2379`. However, `parsedURL.Hostname() := 10.0.0.1`. No matter how many url's are provided in the string, the `Hostname()` method will return the name/address of the first url. I can split then parse, but then it seems like we are duplicating the work of `Hostname()` for our use case. Please let me know if your position stands on this issue and I will update the code.

Additionally, I added the `Hostname() == ""` check because in my testing I found that `Hostname()` usually returns an empty string instead of an error when a bad url is provided.